src/kbmod/reprojection.py

```diff
@@ -159,17 +172,18 @@ def _reproject_work_unit(work_unit, common_wcs, frame="original"):
             mask_add += reprojected_mask
 
         # change all the values where there are is no corresponding data to `KB_NO_DATA`.
-        gaps = footprint_add == 0.0
+        gaps = footprint_add == 0
         science_add[gaps] = KB_NO_DATA
         variance_add[gaps] = KB_NO_DATA
         mask_add[gaps] = 1
 
-        # transforms the mask back into a bitmask.
-        mask_add = np.where(np.isclose(mask_add, 0.0, atol=0.2), 0.0, 1.0)
+        # transforms the mask back into a bitmask. Note that we need to be explicit
+        # about the dtypes for 0.0 and 1.0, otherwise mask_add will be cast to float64.
+        mask_add = np.where(np.isclose(mask_add, 0.0, atol=0.2), np.float32(0.0), np.float32(1.0))
```

Comment:
Without explicitly casting np.float32(0.0) and np.float32(1.0) numpy will recast mask_add to float64. (Another memory copy and extra space required)